project-addons/picking_document/stock.py

```diff
@@ -19,16 +19,35 @@
 #
 ##############################################################################
 
-from openerp import fields, models
-
+from openerp import fields, api, models, _
+import openerp.addons.decimal_precision as dp
 
 class stock_picking(models.Model):
 
     _inherit = 'stock.picking'
 
+
     document_ids = fields.Many2many(
         'stock.document',
         'document_picking_rel',
         'document_id',
         'picking_id',
         'Documents')
+
+    name = fields.Char('name')
+    qty =fields.Integer('qty', compute='_calculate_qty')
+
+    @api.one
+    @api.depends('name')
+    def _calculate_qty(self):
+        picking_name = self.name
+        qty_mid = self.env.cr.execute(
+            """
+SELECT sum(product_uom_qty)
+FROM stock_move
+    INNER JOIN stock_picking ON stock_move.picking_id = stock_picking.id
+WHERE stock_picking.name = '"""+ picking_name + """'"""
+        )
```

Comment:
Nunca debemos usar queries ya que las queries se saltan las reglas multicompañía y no son seguras, tendrías que hacer un bucle for sobre las lineas del albarán el campo se llama "move_lines" e ir sumando en una variable el número de unidades y al final escribes en el campo self.qty la variable.